### ISSUE TITLE
[REVIEW]: Devicely: A Python package for reading, timeshifting and writing sensor data

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3679]
**Submitting author:** @arianesasso (<a href="http://orcid.org/0000-0002-3669-4599">AM Sasso, A Sasso</a>)
**Repository:** <a href="https://github.com/hpi-dhc/devicely/" target ="_blank">https://github.com/hpi-dhc/devicely/</a>
**Version:** v1.1.0
**Editor:** @prashjha
**Reviewer:** @luciorq, @djmitche
**Archive:** Pending

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/3abafc8a04e02d7c61d0bf4fb714af28/status.svg)](https://joss.theoj.org/papers/3abafc8a04e02d7c61d0bf4fb714af28)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/3abafc8a04e02d7c61d0bf4fb714af28"><img src="https://joss.theoj.org/papers/3abafc8a04e02d7c61d0bf4fb714af28/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/3abafc8a04e02d7c61d0bf4fb714af28/status.svg)](https://joss.theoj.org/papers/3abafc8a04e02d7c61d0bf4fb714af28)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/hpi-dhc/devicely/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@luciorq & @djmitche, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @prashjha know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @luciorq
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/hpi-dhc/devicely/">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/hpi-dhc/devicely/">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@arianesasso) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @djmitche
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/hpi-dhc/devicely/">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/hpi-dhc/devicely/">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@arianesasso) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @luciorq, @djmitche it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 1022

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3390/s18061714 is OK
- 10.1038/s41569-021-00522-7 is OK
- 10.1038/s41551-020-00640-6 is OK
- 10.21105/joss.02106 is OK
- 10.21105/joss.01663 is OK
- 10.21105/joss.01867 is OK
- 10.21105/joss.01778 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.23 s (244.1 files/s, 130347.8 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
JavaScript                      14           2405           2473           9213
HTML                             7            998             21           4161
SVG                              1              0              0           2671
Python                          14            399            452           1799
CSS                              4            191             35            757
TeX                              2             40             11            213
Markdown                         4             64              0            190
reStructuredText                 3             50             20             88
Jupyter Notebook                 1              0           2885             66
YAML                             1              8              0             47
TOML                             1              3              0             36
DOS Batch                        1              8              1             26
JSON                             1              0              0             14
make                             1              4              7              9
-------------------------------------------------------------------------------
SUM:                            55           4170           5905          19290
-------------------------------------------------------------------------------


Statistical information for the repository '586ac09723295cb22d15adae' was
gathered on 2021/09/01.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Ariane Morassi Sasso            47           953            662            3.15
Ariane Morassi-Sasso             3            20             19            0.08
Ariane Sasso                     9           568              6            1.12
Jost Morgenstern                 2            12              3            0.03
jostmorgenstern                 64         32444          16585           95.63

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Ariane Morassi Sasso        631           66.2          1.2                6.97
jostmorgenstern           16110           49.7          7.3               15.94
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03679/joss.03679/10.21105.joss.03679.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03679/joss.03679/10.21105.joss.03679.pdf) :page_facing_up: :point_left:

**Comment by prashjha** [comment]
Thank you @djmitche and @luciorq for agreeing to be reviewers. Please read the comments above in this thread and also reviewer guidelines [here](https://joss.readthedocs.io/en/latest/reviewer_guidelines.html). Also, you can browse the closed "REVIEW" issues on the "joss-reviews" repository to get some ideas on how to complete the reviews. Thanks again and good luck!

**Comment by djmitche** [comment]
@arianesasso this looks fantastic.  The contributing guide is particularly well-done.  Just one request: something that helps users figure out how to report a bug and what level of support they can expect.  This may just be a GitHub issue template, for example.

**Comment by arianesasso** [comment]
@djmitche thank you for your feedback! I think that is a great idea, I create a GitHub issue in our repository to address that: https://github.com/hpi-dhc/devicely/issues/32 :)

**Comment by luciorq** [comment]
Great Work for @arianesasso and fellow `devicely` team.

The software is already in a great state and tick most JOSS requirements out of the box.
The authors went the extra mile to enhance the contribution documentation and make the software usable by the broader community.

There are some minimal issues raised in the GitHub repository that I would like to be discussed.

As already linked up here, the issues created during Review are on the [`devicely` Issue Tracker](https://github.com/hpi-dhc/devicely/issues) and are being tracked by the list in this [Meta Issue](https://github.com/hpi-dhc/devicely/issues/37).

Relevant Link: <https://github.com/hpi-dhc/devicely/issues/37>

**Comment by whedon** [comment]
:wave: @luciorq, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @djmitche, please update us on how your review is going (this is an automated reminder).

**Comment by djmitche** [comment]
I think we're both waiting on issues linked above to be solved.

**Comment by arianesasso** [comment]
Dear Reviewers, we are almost done addressing the reviews :).

We will finish them this week, I will write back to you here once we are ready.

**Comment by arianesasso** [comment]
We finished implementing the suggestions :). 
We left the issues open for comments, could you please check them? Thank you!

@djmitche (hpi-dhc/devicely#32) and @luciorq (hpi-dhc/devicely#37)

**Comment by djmitche** [comment]
Will do -- thanks!

**Comment by djmitche** [comment]
The fix in issue 32 looks good to me.  I have no further concerns, having looked through the tracker in issue 37.

**Comment by luciorq** [comment]
Most of my concerns have been discussed, addressed, and updated.
I will be updating/closing the issues during the next day, then will update here!

**Comment by arianesasso** [comment]
Just wanted to quickly check if there are any other open points. Thank you :).

**Comment by djmitche** [comment]
Is there a button I should push for that?

**Comment by luciorq** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by luciorq** [comment]
Dear @prashjha & @arianesasso ,

I just reviewed all commits and closed the relevant issues.
I agree with @djmitche, there aren't any additional considerations. Also, the regenerated manuscript looks good to me.

Therefore, I highly recommend this work for publication.

**Comment by prashjha** [comment]
Super thanks, @luciorq and @djmitche. I will do few checks and then hand it to EiC.

**Comment by prashjha** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3390/s18061714 is OK
- 10.1038/s41569-021-00522-7 is OK
- 10.1038/s41551-020-00640-6 is OK
- 10.21105/joss.02106 is OK
- 10.21105/joss.01663 is OK
- 10.21105/joss.01867 is OK
- 10.21105/joss.01778 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by arianesasso** [comment]
Dear @prashjha we made the corrections and we already have a zenodo DOI: https://doi.org/10.5281/zenodo.5243494 :).

Cheers
Ariane

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5243494" target="_blank">10.5281/zenodo.5243494</a> is the archive.

**Comment by prashjha** [comment]
@arianesasso, could you change the title in zenodo archive to match the title of joss paper?

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1448723258422054922) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2674
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03679](https://doi.org/10.21105/joss.03679)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
This should be updated now. The paper might take ~24 hours to show as updated on the JOSS site due to caching.

**Comment by arianesasso** [comment]
Just saw the update, thanks :).
